google/cloud/bigquery/_helpers.py

```diff
@@ -45,6 +46,15 @@
     re.VERBOSE,
 )
 
+# BigQuery sends INTERVAL data in "canonical format"
+# https://cloud.google.com/bigquery/docs/reference/standard-sql/data-types#interval_type
+_INTERVAL_PATTERN = re.compile(
+    r"(?P<calendar_sign>-?)(?P<years>[0-9]+)-(?P<months>[0-9]+) "
+    r"(?P<days>-?[0-9]+) "
+    r"(?P<time_sign>-?)(?P<hours>[0-9]+):(?P<minutes>[0-9]+):"
+    r"(?P<seconds>[0-9]+)\.?(?P<fraction>[0-9]+)?$"
```

Comment:
Is a lone dot without fractional digits for microseconds possible? In other words, do we accept `H:M:S.` and interpret the missing digits as zero? (which is what Python does, `123.` is acceptable form of `123.0`).

If not, it might make more sense to treat the `.F` part as an atomic optional part, because in the current from, the regex hints that the time part can also be `H:M:SF`, but of course the seconds part will eat the F part, too, since we don't limit the number of matching digits.

Do we know yet what the limits on the number if digits are? I assume the backend does not allow multiple representations of the same interval? (e.g. 125 seconds cannot be expressed as `0:0:125`, but only as `0:2:5`)
(although `relativedelta` can handle that just fine)

Reply:
Actually, that regex has a bigger problem, `fraction` can be present without the dot -- well not really, but it kind of looks like that at first glance and makes me think too hard :).

I think 

`r"(?P<seconds>[0-9]+)(?P<fraction>[.][0-9]*)?`

would be better.

Or even:

`r"(?P<seconds>[0-9]+(?:[.][0-9]*)?)`

then something like (ignoring the time sign):

```
seconds, microseconds = divmod(float(seconds), 1)
micoseconds = int(micoseconds*1000000)
```

which seems simpler to me than the current logic.

Reply:
> Actually, that regex has a bigger problem, `fraction` can be present without the dot -- well not really, but it kind of looks like that at first glance and makes me think too hard :)

Indeed, that was the main point, perhaps I  expressed it in a too convoluted way, sorry. :)

Reply:
`divmod` looks elegant, but I found a rounding problem when I attempted to use this implementation. :-( https://github.com/googleapis/python-db-dtypes-pandas/issues/18

Reply:
>  I assume the backend does not allow multiple representations of the same interval?

When sending data, it seemed to allow it. But when getting data from the API, it always normalized it.